data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/ContentArticlePropertiesUpgrade.java

```diff
@@ -0,0 +1,195 @@
+
+  public static final String       FEATURED_IMAGE_ID                      = "featuredImageId";
+
+  private static final String      FEATURED_IMAGE_UPDATED_DATE            = "featuredImageUpdatedDate";
+
+  public ContentArticlePropertiesUpgrade(InitParams initParams,
+                                         NoteService noteService,
+                                         MetadataService metadataService,
+                                         IdentityManager identityManager,
+                                         SpaceService spaceService,
+                                         UserACL userACL) {
+    super(initParams);
+    this.noteService = noteService;
+    this.metadataService = metadataService;
+    this.identityManager = identityManager;
+    this.spaceService = spaceService;
+    this.userACL = userACL;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    long startupTime = System.currentTimeMillis();
+    LOG.info("Start upgrade of content page properties");
+    int notMigratedContentPagesPropertiesCount;
+    int processedContentPagesPropertiesCount = 0;
+    int totalContentPagesPropertiesCount = 0;
+    try {
+      MetadataFilter metadataFilter = getMetadataFilter();
+      metadataFilter.setMetadataObjectTypes(List.of(NEWS_METADATA_PAGE_VERSION_OBJECT_TYPE,
+                                                    NEWS_METADATA_DRAFT_OBJECT_TYPE,
+                                                    NEWS_METADATA_LATEST_DRAFT_OBJECT_TYPE));
+      List<MetadataItem> metadataItems = metadataService.getMetadataItemsByFilter(metadataFilter, 0, 0);
+      totalContentPagesPropertiesCount = metadataItems.size();
+      for (MetadataItem metadataItem : metadataItems) {
+        if (metadataItem != null && !MapUtils.isEmpty(metadataItem.getProperties())) {
+          Map<String, String> contentProperties = metadataItem.getProperties();
+          Page page = null;
+          String objectType = NOTE_METADATA_PAGE_OBJECT_TYPE;
+          if (metadataItem.getObjectType().equals(NEWS_METADATA_PAGE_VERSION_OBJECT_TYPE)) {
+            PageVersion pageVersion = noteService.getPageVersionById(Long.valueOf(metadataItem.getObjectId()));
+            if (pageVersion != null && pageVersion.getParent() != null) {
+              page = pageVersion.getParent();
+            }
+          } else {
+            page = noteService.getDraftNoteById(metadataItem.getObjectId(), userACL.getSuperUser());
+            objectType = NOTE_METADATA_DRAFT_PAGE_OBJECT_TYPE;
+          }
+          if (page != null) {
+            NoteMetadataObject noteMetadataObject = buildNoteMetadataObject(page, null, objectType);
+            MetadataItem noteMetadataItem = getNoteMetadataItem(page, null, objectType);
+            Map<String, String> noteProperties = new HashMap<>();
+            long creatorId = Long.parseLong(identityManager.getOrCreateUserIdentity(page.getAuthor()).getId());
```

Comment:
We may have some cases where the creator is not defined or has an invalid ID, could you check that case ?

Reply:
Added a check on page.getAuther()
https://github.com/exoplatform/data-upgrade/pull/240/files#diff-f9867c30322aa18037c431dbcf2ff5f6b6abeb32c600f350aeebf11c4aa922ddR123